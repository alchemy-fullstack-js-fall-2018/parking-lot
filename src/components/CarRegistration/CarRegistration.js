import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ROUTES from '../../routes';


export default class CarRegistration extends PureComponent {

  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    history: PropTypes.array
  };

  state = {
    plate: '',
    make: '',
    model: '',
    color: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = event => {
    const { plate, make, model, color } = this.state;
    event.preventDefault();
    this.props.onSubmit({ plate, make, model, color });
    this.props.history.push(ROUTES.CAR.linkTo(plate));
  };

  render() {
    const { plate, make, model, color } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>New Registration</h2>

        <p>
          <label htmlFor="plate">Plate: </label>
          <input type="text" name="plate" value={plate} onChange={this.handleChange} />
        </p>

        <p>
          <label htmlFor="make">Make: </label>
          <input type="text" name="make" value={make} onChange={this.handleChange} />
        </p>

        <p>
          <label htmlFor="model">Model: </label>
          <input type="text" name="model" value={model} onChange={this.handleChange} />
        </p>

        <p>
          <label htmlFor="color">Color: </label>
          <input type="text" name="color" value={color} onChange={this.handleChange} />
        </p>

        <button type="submit">Register</button>
      </form>
    );
  }
}
