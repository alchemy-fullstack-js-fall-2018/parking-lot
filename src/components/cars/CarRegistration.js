import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './CarRegistration.css';
import ROUTES from '../../routes';


export default class CarRegistration extends PureComponent {

  static propTypes = {
    onSubmit: PropTypes.func
  };

  state = {
    plate: '',
    make: '',
    model: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  onSubmit = event => {
    const { onSubmit } = this.props;
    const { plate, make, model } = this.state;

    event.preventDefault();
    onSubmit(plate, make, model);
    this.props.history.push(ROUTES.CAR.linkTo(plate));
  };

  render() {


    return (
      <Fragment>
        <h3 className={styles.header}>Register a New Car</h3>
        <div className={styles.wrapper}>
          <form className={styles.form}>
            <label className={styles.label} htmlFor='plate'>Plate:</label><br/>
            <input id='plate' type='text' name='plate' onChange={this.handleChange} required/><br/>

            <label className={styles.label} htmlFor='make'>Make:</label><br/>
            <input id='make' type='text' name='make' onChange={this.handleChange} required/><br/>

            <label className={styles.label} htmlFor='model'>Model:</label><br/>
            <input id='model' type='text' name='model' onChange={this.handleChange} required/><br/>

            <button className={styles.button} onClick={this.onSubmit}>Add</button>
          </form>
        </div>
      </Fragment>
    );
  }
}

