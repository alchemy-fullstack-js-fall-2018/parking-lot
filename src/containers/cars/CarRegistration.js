import { connect } from 'react-redux';
import { registerCar } from '../../actions/cars';
import CarRegistration from '../../components/CarRegistration/CarRegistration';

const mapStateToProps = () => {
  return { plate: '', make: '', model: '', color: '' };
};

const mapDispatchToProps = dispatch => ({
  onSubmit(car) {
    dispatch(registerCar(car));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarRegistration);
