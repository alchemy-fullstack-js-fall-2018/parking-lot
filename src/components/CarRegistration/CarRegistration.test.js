import React from 'react';
import { shallow } from 'enzyme';
import CarRegistration from '../../components/CarRegistration/CarRegistration';

jest.mock('../../routes.js');

describe ('CarRegistration component', () => {
  let onSubmit = jest.fn();
  let wrapper;
  let car = {
    plate: 'ZZZ000',
    make: 'Plymouth',
    model:  'Edsel',
    color: 'pink'
  };

  beforeEach(() => {
    wrapper = shallow(<CarRegistration onSubmit={onSubmit} history={[]}/>);
  });

  it('renders a CarRegistration component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('submits a car', () => {
    wrapper.find('input').at(0).simulate('change', { target: { name: 'plate', value: car.plate } });
    wrapper.find('input').at(1).simulate('change', { target: { name: 'make', value: car.make } });
    wrapper.find('input').at(2).simulate('change', { target: { name: 'model', value: car.model } });
    wrapper.find('input').at(3).simulate('change', { target: { name: 'color', value: car.color } });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(onSubmit.mock.calls[0][0]).toEqual(car);
  });

});
