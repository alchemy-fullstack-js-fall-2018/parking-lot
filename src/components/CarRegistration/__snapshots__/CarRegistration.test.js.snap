// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarRegistration component renders a CarRegistration component 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    New Registration
  </h2>
  <p>
    <label
      htmlFor="plate"
    >
      Plate: 
    </label>
    <input
      name="plate"
      onChange={[Function]}
      type="text"
      value=""
    />
  </p>
  <p>
    <label
      htmlFor="make"
    >
      Make: 
    </label>
    <input
      name="make"
      onChange={[Function]}
      type="text"
      value=""
    />
  </p>
  <p>
    <label
      htmlFor="model"
    >
      Model: 
    </label>
    <input
      name="model"
      onChange={[Function]}
      type="text"
      value=""
    />
  </p>
  <p>
    <label
      htmlFor="color"
    >
      Color: 
    </label>
    <input
      name="color"
      onChange={[Function]}
      type="text"
      value=""
    />
  </p>
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;
